File: week02/db_query/frontend/src/types/metadata.ts
export type TableType = "table" | "view";

export interface ColumnMetadata {
  columnName: string;
  dataType: string;
  isNullable: boolean;
  isPrimaryKey: boolean;
  defaultValue: string | null;
  position: number;
}

export interface TableMetadata {
  schemaName: string;
  tableName: string;
  tableType: TableType;
  columns: ColumnMetadata[];
}

export interface DatabaseDetail {
  name: string;
  connectionUrl: string;
  createdAt: string;
  updatedAt: string;
  tableCount: number;
  viewCount: number;
  tables: TableMetadata[];
}

File: week02/db_query/backend/src/models/metadata.py
from enum import Enum
from typing import Optional

from src.models import BaseResponseModel


class TableType(str, Enum):
    TABLE = "table"
    VIEW = "view"


class ColumnMetadataResponse(BaseResponseModel):
    column_name: str
    data_type: str
    is_nullable: bool
    is_primary_key: bool
    default_value: Optional[str] = None
    position: int


class TableMetadataResponse(BaseResponseModel):
    schema_name: str
    table_name: str
    table_type: TableType
    columns: list[ColumnMetadataResponse]


class DatabaseDetailResponse(BaseResponseModel):
    name: str
    connection_url: str
    created_at: str
    updated_at: str
    table_count: int
    view_count: int
    tables: list[TableMetadataResponse]
